RIOT-OS/RIOT
# drivers/n25q128: driver implementation for the n25q128 serial flash memory

### Contribution description
Initial driver implementation for the [N25Q128](https://datasheet4u.com/datasheet-pdf/NumonyxB.V/N25Q128/pdf.php?id=683085) serial flash memory device, which is also installed on the [IoT-LAB M3 open node](https://doc.riot-os.org/group__boards__iotlab-m3.html).

The driver will support the basic ['Extended SPI'](https://html.alldatasheet.com/html-pdf/353314/NUMONYX/N25Q128/6294/21/N25Q128.html)-protocol and all [instructions](https://html.alldatasheet.com/html-pdf/353314/NUMONYX/N25Q128/23394/78/N25Q128.html), except those intended for the 'Dual I/O SPI'- and 'Quad I/O SPI'-protocol.




### Testing procedure
Check the instructions for functionality. For example, check if the bytes read from the ID-instruction match with the bytes in the data sheet.




### Issues/PRs references
